src/game/emutest.h

```diff
@@ -0,0 +1,40 @@
+#ifndef EMUTEST_H
+#define EMUTEST_H
+
+enum Emulator {
+    EMU_WIIVC = 0x0001,
+    EMU_PROJECT64_ANY = 0x001E,
+        EMU_PROJECT64_1_OR_2 = 0x0006, // PJ64 1.6 or similar (up to 2.3)
+        EMU_PROJECT64_3 = 0x0008, // PJ64 3.X
+        EMU_PROJECT64_4 = 0x0010, // PJ64 4.0 or later
+    EMU_MUPEN_BASED = 0x0060, // mupen64plus or pre-2.12 paralleln64, but NOT simple64 or new paralleln64
+        EMU_MUPEN_OLD = 0x0020, // Nemu64 and pre-2.12 paralleln64 will also get detected as this
+        EMU_MUPEN64PLUS_NEXT = 0x040,
+    EMU_CEN64 = 0x0080,
+    EMU_SIMPLE64 = 0x0100,
+    EMU_PARALLELN64 = 0x0200, // ParallelN64 2.12 or later
+    EMU_ARES = 0x0400,
+    EMU_CONSOLE = 0x0800
+};
+
+// initializes gEmulator
+extern void detect_emulator();
+
+/* gEmulator is an enum that identifies the current emulator.
+ * The enum values work as a bitfield, so you can use the & and | operators
+ * to test for multiple emulators or versions at once.
+ * 
+ * Examples:
+ * 
+ * Test for any version of PJ64:
+ * if (gEmulator & EMU_PROJECT64_ANY)
+ * 
+ * Test for only PJ64 < 3.0:
+ * if (gEmulator & EMU_PROJECT64_1_OR_2)
+ * 
+ * Test for Console, Ares, or ParallelN64:
+ * if (gEmulator & (EMU_CONSOLE | EMU_ARES | EMU_PARALLELN64))
+ */
+extern enum Emulator gEmulator;
+
```

Comment:
for backwards compatibility, can you add: 
`#define gIsConsole (gEmulator & EMU_CONSOLE)`

Reply:
Doesn't work for `if (gIsConsole == TRUE)`; in that way, I'd argue the compiler issues are actually safer than not.

Reply:
I suppose that could work too

Reply:
yeah i like that falco

Reply:
Do you want `gIsVC` as well?

Reply:
That one is much less necessary; I doubt we have a lot of people actively supporting VC compared to console

Reply:
yeah ^ exactly

Reply:
Done